torch/csrc/Storage.cpp

```diff
@@ -25,8 +25,14 @@
 #include "generic/Storage.cpp"
 #include <TH/THGenerateHalfType.h>
 
+#ifndef USE_CUDA
+// NB: When USE_CUDA, the *full* implementation of
```

Comment:
can you just put the full definition here with ifdefs in the definition?  It's much nicer to have it in one place (and there's precedent for putting stuff like that at this directory level, even though it would be nicer if this level was reserved for cross cpu/cuda things, and all the cpu specific things were moved to /cpu).